### PR TITLE
Merge sync-imager branch

### PR DESCRIPTION
**Pull request by vidurb**
Contains the addition of the Imager plugin, relevant configuration and the _image component.

### CODE REVIEW

File: config/imager.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                'path' => '/usr/bin/jpegoptim',
+            ],
+            'optipng' => [
+                'extensions' => ['png'],
+                'path' => '/usr/bin/optipng',
+                'optionString' => '-o2',
+            ],
+        ],
+
+        'clearKey' => 'SH0ZDR%PVbqwg%k7lMt&t0tyHWRp0Bh@',
```
Comment (rungta) [inline config/imager.php]: This value is supposed to be unique and secret for each project. Should probably disable (comment out) for the boilerplate.

File: templates/_components/_image.twig [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+{# Variables #}
+{% set queries = {
+  '--sm-plus':  '(min-width:' ~ 576 ~ 'px)',
+  '--md-plus':  '(min-width:' ~ 768 ~ 'px)',
+  '--lg-plus':  '(min-width:' ~ 992 ~ 'px)',
+  '--xl-plus':  '(min-width:' ~ 1200 ~ 'px)',
+  '--retina':   '(min-resolution: 192dpi)',
+} %}
+{% set presets = {} %}
+{# {% set imageAsset = imageAsset and not craft.embeddedAssets.isEmbedded(imageAsset) ? imageAsset : false %} #}
```
Comment (rungta) [inline templates/_components/_image.twig]: We can get rid of this line and any other instances of `craft.embeddedAssets`

**Review by rungta — CHANGES_REQUESTED** [review]

File: config/imager.php (hunk 1 above)
Comment (vidurb) [reply config/imager.php]: Oh, okay. It was originally the same for the last project we did and the second last project, so I thought you considered it unnecessary to change it.

File: templates/_components/_image.twig (hunk 2 above)
Comment (vidurb) [reply templates/_components/_image.twig]: Removed that line in the next to next commit.

File: config/imager.php (hunk 1 above)
Comment (vidurb) [reply config/imager.php]: Commented out that line in the next commit.

**Review by rungta — APPROVED** [review]

**Pull request merged**